src/Vault.sol

```diff
@@ -783,11 +809,45 @@ contract Vault {
         }
     }
 
+    /*///////////////////////////////////////////////////////////////
+                    ASSET MANAGEMENT LOGIC
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Calls external action handlers to execute and interact with external logic.
+     * @dev Similar to flash loans, this function optimistically calls external logic and checks for the vault state at the very end.
+     * @param actionHandler the address of the action handler to call
+     * @param actionData a bytes object containing two actionAssetData structs, an address array and a bytes array
+     */
+    function vaultManagementAction(address actionHandler, bytes calldata actionData) public onlyOwner {
```

Comment:
Not in scope of this PR, but maybe a new ticket, should this be onlyOwner or should we work with permissions (-> keeper networks can trigger certain actions)

Reply:
#177